frontend/src/devicelist/__tests__/DeviceListPageContents.test.jsx

```diff
@@ -30,167 +33,232 @@ describe("<DeviceListPageContents/> class component", () => {
   });
 
   describe("componentDidMount() function", () => {
+    beforeEach(() => {
+      wrapper = Enzyme.shallow(<DeviceListPageContents />, {
+        disableLifecycleMethods: true
+      });
+    });
     it("calls DeviceApi.getSenders() and getReceivers()", () => {
       jest.spyOn(DeviceApi, "getSenders").mockImplementation(() => {
-        return dummySenders;
+        return Promise.resolve(dummySenders);
       });
       jest.spyOn(DeviceApi, "getReceivers").mockImplementation(() => {
-        return dummyReceivers;
-      });
-      wrapper = Enzyme.shallow(<DeviceListPageContents />, {
-        disableLifecycleMethods: true
+        return Promise.resolve(dummyReceivers);
       });
+
       wrapper.instance().componentDidMount();
 
-      expect(DeviceApi.getSenders).toHaveBeenCalledWith(
-        wrapper.instance().handleSendersChange
-      );
-      expect(DeviceApi.getReceivers).toHaveBeenCalledWith(
-        wrapper.instance().handleReceiversChange
-      );
+      expect(DeviceApi.getSenders).toHaveBeenCalled();
+      expect(DeviceApi.getReceivers).toHaveBeenCalled();
     });
-  });
-  describe("handleChange()", () => {
-    beforeEach(() => {
-      wrapper = Enzyme.shallow(<DeviceListPageContents />);
-    });
-    it("should set the state", () => {
-      const startingState = {
-        selected: 0,
-        senders: [],
-        receivers: []
+    it("if getSenders() rejects, an error snackbar with the caught error message is displayed", async () => {
+      const returnedError = {
+        message: "test"
       };
+      jest.spyOn(DeviceApi, "getSenders").mockRejectedValue(returnedError);
+      jest.spyOn(DeviceApi, "getReceivers").mockImplementation(() => {
+        return Promise.resolve(dummyReceivers);
+      });
```

Comment:
Replace with `mockResolvedValue` (and the one in the next test)

Reply:
I tried this before and the tests failed using `mockResolvedValue` for some odd reason.

Reply:
Actually I have no idea why but that works again (mockResolveValue) so I'll be changing the tests to reflect that.

Reply:
cc0b92d